fe/estimator.py

```diff
@@ -9,16 +9,23 @@
 
 from typing import Tuple, List, Any
 
+import dataclasses
+import jax.numpy as jnp
+
+@dataclasses.dataclass
 class SimulationResult:
-    def __init__(self, xs=None, du_dls=None, du_dps=None):
-        self.xs = xs
-        self.du_dls = du_dls
-        self.du_dps = du_dps
+   xs: np.array
+   du_dls: np.array
+   du_dps: np.array
 
-    def save(self, path):
-        return np.savez(path, xs=self.xs, du_dls=self.du_dls, du_dps=self.du_dps)
+def flatten(v):
+  return tuple(), (v.xs, v.du_dls, v.du_dps)
 
+def unflatten(aux_data, children):
+  xs, du_dls, du_dps = aux_data
+  return SimulationResult(xs, du_dls, du_dps)
```

Comment:
Formatting: these few lines may have inconsistent indentation with the rest of the file

Reply:
willfix